src/contexts/SiteLaunchContext.tsx

```diff
@@ -0,0 +1,86 @@
+import { createContext, useContext, useEffect, useState } from "react"
+import { useParams } from "react-router-dom"
+
+import { useGetSiteLaunchStatus } from "hooks/siteDashboardHooks"
+
+import {
+  SiteLaunchDto,
+  SiteLaunchFrontEndStatus,
+  SiteLaunchStatusProps,
+  SITE_LAUNCH_TASKS_LENGTH,
+} from "types/siteLaunch"
+
+interface SiteLaunchContextProps {
+  siteLaunchStatusProps: SiteLaunchStatusProps
+  setSiteLaunchStatusProps: (
+    siteLaunchStatusProps: SiteLaunchStatusProps
+  ) => void
+}
+
+const SiteLaunchContext = createContext<SiteLaunchContextProps | null>(null)
+
+export const useSiteLaunchContext = (): SiteLaunchContextProps => {
+  const SiteLaunchContextData = useContext(SiteLaunchContext)
+  if (!SiteLaunchContextData)
+    throw new Error("useSiteLaunchContext must be used within an RoleProvider")
+  return SiteLaunchContextData
+}
+
+export const SiteLaunchProvider = ({
+  children,
+  initialSiteLaunchStatus = "NOT_LAUNCHED",
+  initialStepNumber = 0,
+}: {
+  children: React.ReactNode
+  initialSiteLaunchStatus?: SiteLaunchFrontEndStatus
+  initialStepNumber?: number
+}): JSX.Element => {
+  const { siteName } = useParams<{ siteName: string }>()
+
+  const [
+    siteLaunchStatusProps,
+    setSiteLaunchStatusProps,
+  ] = useState<SiteLaunchStatusProps>({
+    siteLaunchStatus: initialSiteLaunchStatus,
+    stepNumber: initialStepNumber,
+  })
+  const { data: siteLaunchDto } = useGetSiteLaunchStatus(siteName)
+  const { siteStatus: siteStatusDto, stepNumber: stepNumberDto } =
+    siteLaunchDto ||
+    ({
+      siteStatus: "NOT_LAUNCHED", // defaulting to not launched
+      stepNumber: 0,
+    } as SiteLaunchDto)
+  useEffect(() => {
```

Comment:
`siteLaunchStatusProps` seems to be a mapping from `siteStatusDto` -> `siteLaunchStatusProps`, which would be clearer as a function (and we can shift it out of the component) - the `useEffect` should ideally perform some effect (as the name implies) like data fetching/mutation on our backend.

Reply:
thanks for this man, I knew that `useEffect` is used to sync external to internal flow, but here I tried using it to "sync" two hooks which the `useEffect` shoudnt be used for. 
Abstracted this out for clarity as suggested, let me know if this makes sense!

Reply:
kek turns out is a `useEffect` after all